### PR TITLE
Mc/tcs premium auction

### PR DESCRIPTION
**Pull request by microwavedcola1**

### CODE REVIEW

File: ts/client/src/client.ts [hunk 1]
```diff
@@ -4339,9 +4325,11 @@ export class MangoClient {
           1) *
         100;
     }
-    const maxPricePremiumRate =
+    let maxPricePremiumRate =
       maxPricePremiumPercent > 0 ? maxPricePremiumPercent / 100 : 0.03;
 
+    maxPricePremiumRate = maxPricePremiumRate * 2;
```
Comment (ckamm) [inline ts/client/src/client.ts:4331]: Maybe make a separate helper for the 60s duration, 2x premium rate thing; I don't think this should be in this function

File: ts/client/src/client.ts [hunk 2]
```diff
@@ -3756,8 +3752,9 @@ export class MangoClient {
     );
     const lowerLimit = 0;
     const upperLimit = thresholdPriceNativeNative;
+    console.log(thresholdPriceNativeNative);
```
Comment (ckamm) [inline ts/client/src/client.ts:3755]: logging

File: ts/client/src/client.ts [hunk 3]
```diff
@@ -4339,8 +4339,10 @@ export class MangoClient {
           1) *
         100;
     }
-    const maxPricePremiumRate =
+    let maxPricePremiumRate =
       maxPricePremiumPercent > 0 ? maxPricePremiumPercent / 100 : 0.03;
+    maxPricePremiumRate =
+      maxPricePremiumRate * premiumMultiplier + extraPricePremiumBps / 10000;
```
Comment (ckamm) [inline ts/client/src/client.ts:4345]: Imo this modification of user-specified premium is weird. Would prefer if this was just done externally to the price premium argument. This function should just build the instruction with the passed-in args.

Comment (microwavedcola1) [reply ts/client/src/client.ts:4345]: They do have defaults which act as no-op

**Pull request merged**